src/utils/errors.rs

```diff
@@ -0,0 +1,114 @@
+use std::fmt;
+use thiserror::Error;
+use rocket::request::Request;
+use rocket::response::{self, Response, Responder};
+use rocket::http::{ContentType, Status};
+use serde_json;
+use serde::{Serialize, Deserialize};
+use std::io::Cursor;
+use log::debug;
+use anyhow::Result;
+
+pub type ApiResult<T, E = ApiError> = Result<T, E>;
+
+#[derive(Error, Debug, Serialize)]
+pub struct ApiError {
+    pub status: u16,
+    pub details: ErrorDetails,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct ErrorDetails {
+    pub code: u64,
+    pub message: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub arguments: Option<Vec<String>>,
+}
+
+impl ApiError {
+    pub fn from_backend_error(status_code: u16, raw_error: &str) -> ApiError {
+        match serde_json::from_str::<ErrorDetails>(&raw_error) {
+            Ok(backend_error) => {
+                ApiError::new(status_code, backend_error)
+            }
+            Err(error) => {
+                let error_details = ErrorDetails {
+                    code: 1337,
```

Comment:
I would make this an error code `42` and return the `raw_error` as the message (not the serde error)